packages/context/src/interceptor-chain.ts

```diff
@@ -131,40 +145,32 @@ export class GenericInterceptorChain<C extends Context = Context> {
   /**
    * Invoke the interceptor chain
    */
-  invokeInterceptors(): ValueOrPromise<InvocationResult> {
+  invokeInterceptors(): ValueOrPromise<T | undefined> {
```

Comment:
As commented earlier, I think we should not be adding `undefined` to the retval-type provided by the user and use `ValueOrPromise<T>`.